Firestore/core/src/remote/firebase_metadata_provider_apple.mm

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/remote/firebase_metadata_provider_apple.h"
+
+#import "FirebaseCore/Sources/Private/FIRAppInternal.h"
+#import "FirebaseCore/Sources/Private/FIRHeartbeatInfo.h"
+#import "FirebaseCore/Sources/Private/FIROptionsInternal.h"
+
+#include "Firestore/core/src/util/string_apple.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+namespace {
+
+using util::MakeString;
+
+std::string GetUserAgent() {
+  return MakeString([FIRApp firebaseUserAgent]);
+}
+
+std::string GetHeartbeat() {
+  return std::to_string([FIRHeartbeatInfo heartbeatCodeForTag:@"fire-fst"]);
+}
+
+std::string GetGmpAppId(FIRApp* app) {
+  return MakeString(app.options.googleAppID);
+}
+
+}  // namespace
+
+FirebaseMetadataProviderApple::FirebaseMetadataProviderApple(FIRApp* app)
+    : app_(app) {
+}
+
+void FirebaseMetadataProviderApple::UpdateMetadata(
+    grpc::ClientContext& context) {
+  context.AddMetadata(kXFirebaseClientHeader, GetUserAgent());
+  context.AddMetadata(kXFirebaseClientLogTypeHeader, GetHeartbeat());
```

Comment:
I don't think the headers are sent unconditionally on android:
https://github.com/firebase/firebase-android-sdk/blob/6e286da45f952142d91482219f0ad1e019c2b06d/firebase-firestore/src/main/java/com/google/firebase/firestore/remote/FirebaseClientGrpcMetadataProvider.java#L55

cc @VinayGuthal to confirm

Reply:
Great point, thanks. Before I do the change, one thing I need to clarify is whether the GMP App ID should or shouldn't be sent if heartbeat is `0`. @VinayGuthal

Reply:
I think sending the user agent, heartbeat type and GMP App ID may be omitted when the heartbeat type is `FIRHeartbeatInfoCodeNone` if the headers are meant to be used for the Platform logging pipeline only as it won't affect instance count estimation. Though I see a potential value in sending the user agent anyway as it  may be used for analysis potential SDK issues.

Reply:
hmm the concern at the time was that we might be causing a latency everytime firestore is used and hence the headers were not sent unconditionally. As @maksymmalyhin mentioned it won't affect instance count estimation.

Reply:
Thanks for the explanations. Changed to not send anything if the heartbeat is `0` (how often does it happen, by the way?), PTAL.

Reply:
the heartbeat is 0 all the time except for the first time in the day.